lib/active_model/serializer/adapter/json_api/deserialization.rb

```diff
@@ -0,0 +1,73 @@
+module ActiveModel
+  class Serializer
+    module Adapter
+      class JsonApi
+        module Deserialization
+          module_function
+
+          # Parse a Hash or ActionController::Parameters representing a JSON API document
+          # into an ActiveRecord-ready hash.
+          #
+          # @param [Hash|ActionController::Parameters] document
+          # @param [Hash] options
+          #   fields: Array of symbols and a Hash. Specify whitelisted fields, optionally
+          #      specifying the attribute name on the model.
+          # @return [Hash] ActiveRecord-ready hash
+          #
+          def parse(document, options = {})
```

Comment:
I think calling from_json on json-api data should generate a model / associations / whatever.

so, imo, from_json could call this method, `parse` and return an instance of a model.